core/src/main/java/io/bisq/core/btc/wallet/WalletsSetup.java

```diff
@@ -300,35 +300,41 @@ private void configPeerNodesForLocalHostBitcoinNode() {
 
     private void configPeerNodes(Socks5Proxy socks5Proxy) {
         List<BitcoinNodes.BtcNode> btcNodeList = new ArrayList<>();
-        walletConfig.setMinBroadcastConnections((int) Math.floor(DEFAULT_CONNECTIONS * 0.8));
         switch (BitcoinNodes.BitcoinNodesOption.values()[preferences.getBitcoinNodesOptionOrdinal()]) {
             case CUSTOM:
-                String bitcoinNodesString = preferences.getBitcoinNodes();
-                if (bitcoinNodesString != null) {
+                final String bitcoinNodesString = preferences.getBitcoinNodes();
+                if (bitcoinNodesString != null && !bitcoinNodesString.isEmpty()) {
                     btcNodeList = Splitter.on(",")
                             .splitToList(StringUtils.deleteWhitespace(bitcoinNodesString))
                             .stream()
                             .filter(e -> !e.isEmpty())
                             .map(BitcoinNodes.BtcNode::fromFullAddress)
                             .collect(Collectors.toList());
+                    walletConfig.setMinBroadcastConnections((int) Math.ceil(btcNodeList.size() * 0.5));
+                    // If Tor is set we usually only use onion nodes, but if user provides mixed clear net and onion nodes we want to use both
+                    useAllProvidedNodes = true;
```

Comment:
It looks like the semantics of `useAllProvidedNodes` is getting overloaded here, because in this context, it really means "use all _custom_ nodes" that have been specified by the user at the command line. That'll be pretty confusing for anyone who doesn't have the context of this pull request.

It's also generally unfortunate to introduce this kind of mutability, but for a quick fix, at least consider changing the name to something like `useAllSpecifiedNodes`, so it at least catches the "provided" and "custom" cases equally.

Reply:
I added a new flag.